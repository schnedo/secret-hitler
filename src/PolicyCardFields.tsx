import { ReactElement } from "react";
import { useSelector } from "react-redux";
import { RootState } from "./store";

export default function PolicyCardFields(): ReactElement {
  const policyDeck = useSelector((state: RootState) => state.policyDeck);
  return (
    <div>
      <div>size draw pile: {policyDeck.drawingPile.length}</div>
      <div>size discard pile: {policyDeck.discardPile.length}</div>
      <div>liberal policies enacted: {policyDeck.nLiberalsPlayed}</div>
      <div>fascist policies enacted: {policyDeck.nFascistsPlayed}</div>
    </div>
  );
}
