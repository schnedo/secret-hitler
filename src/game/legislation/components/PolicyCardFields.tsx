import type { ReactElement } from "react";
import type PolicyDeck from "../PolicyDeck";

export interface PolicyCardFieldsProps {
  drawingPile: PolicyDeck;
  discardPile: PolicyDeck;
  nLiberalsPlayed: number;
  nFascistsPlayed: number;
}

export default function PolicyCardFields({
  drawingPile,
  discardPile,
  nLiberalsPlayed,
  nFascistsPlayed,
}: PolicyCardFieldsProps): ReactElement {
  return (
    <div>
      <div>size draw pile: {drawingPile.length}</div>
      <div>size discard pile: {discardPile.length}</div>
      <div>liberal policies enacted: {nLiberalsPlayed}</div>
      <div>fascist policies enacted: {nFascistsPlayed}</div>
    </div>
  );
}
