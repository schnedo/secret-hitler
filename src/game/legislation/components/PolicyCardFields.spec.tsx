import { render } from "@testing-library/react";
import PolicyCardFields from "./PolicyCardFields";

it("should render correctly", async () => {
  expect.hasAssertions();

  const { container } = render(
    <PolicyCardFields
      drawingPile={[]}
      discardPile={[]}
      nLiberalsPlayed={1}
      nFascistsPlayed={3}
    />,
  );

  expect(container).toMatchSnapshot();
});
