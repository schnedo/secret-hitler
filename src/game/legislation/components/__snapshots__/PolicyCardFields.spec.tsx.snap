// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <div>
      size draw pile: 
      0
    </div>
    <div>
      size discard pile: 
      0
    </div>
    <div>
      liberal policies enacted: 
      1
    </div>
    <div>
      fascist policies enacted: 
      3
    </div>
  </div>
</div>
`;
